Symfony/CS/Tests/Fixer/Contrib/PhpUnitDedicateAssertFixerTest.php

```diff
@@ -0,0 +1,206 @@
+                    $this->assertFalse(\is_null($a), "my message");
+                ',
+            ),
+            array(
+                '<?php
+                    $this->assertEmpty($a);
+                    $this->assertNotEmpty($a);
+                ',
+                '<?php
+                    $this->assertTrue(empty($a));
+                    $this->ASSERTFALSE(empty($a));
+                ',
+            ),
+            array(
+                '<?php
+                    $this->assertInfinite($a);
+                    $this->assertFinite($a, "my message");
+                    $this->assertInfinite($a);
+                    $this->assertFinite($a, "my message");
+                ',
+                '<?php
+                    $this->assertTrue(is_infinite($a));
+                    $this->assertFalse(is_infinite($a), "my message");
+                    $this->assertTrue(\is_infinite($a));
+                    $this->assertFalse(\is_infinite($a), "my message");
+                ',
+            ),
+            array(
+                '<?php
+                    $this->assertArrayHasKey("test", $a);
+                    $this->assertArrayNotHasKey($b, $a, $c);
+                ',
+                '<?php
+                    $this->assertTrue(\array_key_exists("test", $a));
+                    $this->ASSERTFALSE(array_key_exists($b, $a), $c);
+                ',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider provideNotFixCases
+     */
+    public function testNotFix($expected)
+    {
+        $this->makeTest($expected);
+    }
+
+    public function provideNotFixCases()
+    {
+        return array(
+            array(
+                '<?php echo $this->assertTrue;',
+            ),
+            array(
+                '<?php echo $this->assertTrue?>',
+            ),
+            array(
+                '<?php
+                    const is_null = 1;
+                    $this->assertTrue(is_null);
+                    $this->assertTrue(is_int($a) && $b);
+                    $this->assertFalse(is_nan($a));
+                    $this->assertTrue(is_int($a) || \is_bool($b));
+                    $this->assertTrue($a&&is_int($a));
+                ',
+            ),
+        );
+    }
+
+    public function testConfig()
+    {
+        $fixer = $this->getFixer();
+        $fixer->configure(array('file_exists'));
+        $this->makeTest(
+            '<?php
+                    $this->assertFileExists($a);
+                    $this->assertTrue(is_infinite($a));
+            ',
+            '<?php
+                    $this->assertTrue(file_exists($a));
+                    $this->assertTrue(is_infinite($a));
+            ',
+            null,
+            $fixer
+        );
+    }
+
+    /**
+     * @expectedException \Symfony\CS\ConfigurationException\InvalidFixerConfigurationException
+     * @expectedExceptionMessage [php_unit_dedicate_assert] Unknown configuration method "_unknown_".
```

Comment:
Please always use `@expectedExceptionMessageRegExp`.
`@expectedExceptionMessage foo`
means
`@expectedExceptionMessageRegExp foo`
which is
`@expectedExceptionMessageRegExp ^.*foo.*$`
and not what we want, that is:
`@expectedExceptionMessageRegExp ^foo$`